### PR TITLE
feat: Add school timetabling quickstart for Python

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
## Description of the change

> Description here

### Development

- [ ] The changes have been covered with tests, if necessary.
- [ ] You have a green build, with the exception of the flaky tests.
- [ ] UI and JS files are fully tested, the user interface works for all modules affected by your changes (e.g., solve and analyze buttons).
- [ ] The network calls work for all modules affected by your changes (e.g., solving a problem).
- [ ] The console messages are validated for all modules affected by your changes.

### Code Review

- [ ] This pull request includes an explanatory title and description.
- [ ] The GitHub issue is linked.
- [ ] At least one other engineer has approved the changes.
- [ ] After PR is merged, inform the reporter.

### CODE REVIEW

File: python/school-timetabling/static/app.js [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+                <button class="nav-link" id="navRest" data-bs-toggle="pill" data-bs-target="#rest" type="button">Guide</button>
+              </li>
+              <li class="nav-item" id="navOpenApiItem">
+                <button class="nav-link" id="navOpenApi" data-bs-toggle="pill" data-bs-target="#openapi" type="button">REST API</button>
+              </li>
+            </ul>
+          </div>
+          <div class="ms-auto">
+              <div class="dropdown">
+                  <button class="btn btn-secondary dropdown-toggle" type="button" id="dropdownMenuButton" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
```
Comment (zepfred) [inline python/school-timetabling/static/app.js:408]: This button is not working. Let's fix it.

**Comment by zepfred** [comment]
@Christopher-Chianelli, I'm getting the following when using Python 3.10:

```
INFO:     @ uvicorn.error : Started server process [159046]
INFO:     @ uvicorn.error : Waiting for application startup.
INFO:     @ uvicorn.error : Application startup complete.
INFO:     @ uvicorn.error : Uvicorn running on http://127.0.0.1:8080 (Press CTRL+C to quit)
INFO:     @ uvicorn.access : 127.0.0.1:38646 - "GET / HTTP/1.1" 304
INFO:     @ uvicorn.access : 127.0.0.1:38646 - "GET /webjars/timefold/css/timefold-webui.css HTTP/1.1" 304
INFO:     @ uvicorn.access : 127.0.0.1:38656 - "GET /app.js HTTP/1.1" 304
INFO:     @ uvicorn.access : 127.0.0.1:38648 - "GET /webjars/timefold/js/timefold-webui.js HTTP/1.1" 304
INFO:     @ uvicorn.access : 127.0.0.1:38648 - "GET /q/swagger-ui HTTP/1.1" 200
INFO:     @ uvicorn.access : 127.0.0.1:38648 - "GET /webjars/timefold/img/timefold-logo-horizontal-negative.svg HTTP/1.1" 304
INFO:     @ uvicorn.access : 127.0.0.1:38648 - "GET /demo-data HTTP/1.1" 200
INFO:     @ uvicorn.access : 127.0.0.1:38648 - "GET /demo-data/SMALL HTTP/1.1" 200
INFO:     @ uvicorn.access : 127.0.0.1:38648 - "GET /openapi.json HTTP/1.1" 200
INFO:     @ uvicorn.access : 127.0.0.1:38648 - "GET /webjars/timefold/img/timefold-favicon.svg HTTP/1.1" 304
INFO:     @ uvicorn.access : 127.0.0.1:38648 - "POST /timetables HTTP/1.1" 200
INFO:     @ timefold.solver : Solving started: time spent (44), best score (-40init/0hard/0soft), environment mode (REPRODUCIBLE), move thread count (NONE), random (JDK with seed 0).
INFO:     @ timefold.solver : Problem scale: entity count (20), variable count (40), approximate value count (13), approximate problem scale (3.486744 × 10^29).
INFO:     @ uvicorn.access : 127.0.0.1:38648 - "GET /timetables/ID HTTP/1.1" 500
ERROR:    @ uvicorn.error : Exception in ASGI application
Traceback (most recent call last):
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 399, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__
    return await self.app(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/starlette/applications.py", line 123, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/starlette/middleware/errors.py", line 186, in __call__
    raise exc
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/starlette/middleware/errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/starlette/middleware/exceptions.py", line 65, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 756, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 776, in app
    await route.handle(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 297, in handle
    await self.app(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 77, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 72, in app
    response = await func(request)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/fastapi/routing.py", line 278, in app
    raw_response = await run_endpoint_function(
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/.venv/lib64/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/school-timetabling/src/school_timetabling/routes.py", line 43, in get_timetable
    timetable = data_sets[problem_id]
KeyError: 'ID'
INFO:     @ timefold.solver : Construction Heuristic phase (0) ended: time spent (4173), best score (-1hard/-15soft), score calculation speed (145/sec), step total (20).
```

Could you confirm if there is a problem with the logic or if something is wrong on my end?

File: python/school-timetabling/tests/test_feasible.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from timefold.solver import SolverFactory
+from timefold.solver.config import (SolverConfig, ScoreDirectorFactoryConfig,
+                                    TerminationConfig, Duration, TerminationCompositionStyle)
+
+from school_timetabling.domain import Timetable, Lesson
+from school_timetabling.constraints import school_timetabling_constraints
+from school_timetabling.demo_data import generate_demo_data, DemoData
+
+
+def test_feasible():
```
Comment (zepfred) [inline python/school-timetabling/tests/test_feasible.py:11]: Let's either create a new test or update this one to enable API calls, allowing us to test the API with supported Python versions.

**Review by zepfred — CHANGES_REQUESTED** [review]
The logic looks good. However, I want to ensure it runs on the supported Python versions.

Concerning testing the API, this also applies to `employee-scheduling` module, as we could configure the CI to test them using all supported Python versions.

**Review by zepfred — APPROVED** [review]
Looks good. I've confirmed it works on Python 3.10 now. However, I still think 'test_feasible.py' should make API calls as we do in Java. Using API calls during the tests can help to catch issues like the one that occurred on my end.

**Comment by triceo** [comment]
@Christopher-Chianelli I think @zepfred has a point. Please update the test.

**Pull request merged**